pkgs/tools/misc/geekbench/4.nix

```diff
@@ -12,23 +19,21 @@ stdenv.mkDerivation rec {
   dontConfigure = true;
   dontBuild = true;
 
-  nativeBuildInputs = [ makeWrapper ];
+  nativeBuildInputs = [ autoPatchelfHook makeWrapper ];
+  buildInputs = [ stdenv.cc.cc ];
 
   installPhase = ''
-    mkdir -p $out/bin $out/lib
-    cp -r geekbench.plar geekbench4 geekbench_x86_64 $out/bin
+    runHook preInstall
 
-    # needed for compute benchmark
-    ln -s ${linuxPackages.nvidia_x11}/lib/libcuda.so $out/lib/
-    ln -s ${ocl-icd}/lib/libOpenCL.so $out/lib/
-    ln -s ${ocl-icd}/lib/libOpenCL.so.1 $out/lib/
-    ln -s ${vulkan-loader}/lib/libvulkan.so $out/lib/
-    ln -s ${vulkan-loader}/lib/libvulkan.so.1 $out/lib/
+    mkdir -p $out/bin
+    cp -r geekbench.plar geekbench4 geekbench_x86_64 $out/bin
 
     for f in geekbench4 geekbench_x86_64 ; do
-      patchelf --set-interpreter $(cat ${stdenv.cc}/nix-support/dynamic-linker) $out/bin/$f
-      wrapProgram $out/bin/$f --prefix LD_LIBRARY_PATH : "${lib.makeLibraryPath [ stdenv.cc.cc.lib ]}:$out/lib/"
+      wrapProgram $out/bin/$f \
+        --prefix LD_LIBRARY_PATH : "${lib.makeLibraryPath [ ocl-icd vulkan-loader ]}:/run/opengl-driver/lib"
```

Comment:
There is `addOpenGLRunpath`.

Reply:
I tried it before but it interact badly with `wrapProgram`.

Reply:
It should work if invoked before wrapProgram.

Reply:
I did, the program builds successfully but it doesn't find CUDA. If I remove the wrap CUDA works but not the Vulcan/OpenCL.

Reply:
Then at least we can use `addOpenGLRunpath.driverLink` instead of hardcoding `/run/opengl-driver`? The path is subject to change due to https://github.com/NixOS/rfcs/pull/121

Reply:
Done. Thanks for the suggestion.